### PR TITLE
Major syntax changes

### PR DESCRIPTION
**Pull request by jsn1993**
This PR implements four important syntax changes:

- `__call__` connection is deprecated. This is somewhat convenient but the keyword argument approach doesn't work for cases when we need to connect individual ports in the interface.
- Signal now only takes Bits or bitstruct type. We used to allow arbitrary type but throughout the practice we don't find creating Wire(int) very useful.
- Signal now allows taking a single integer N as a shortcut for mk_bits(N). This might avoid a lot of mk_bits calls ... 
- Support `@update` and @update_ff. Dropping @s.update.

### CODE REVIEW

File: examples/ex03_proc/ProcRTL.py [hunk 1]
```diff
@@ -53,112 +53,110 @@ def construct( s ):
...
-      imemreq_addr  = s.imemreq_q.enq.msg.addr,
-      imemresp_data = s.imemresp_drop.out.ret,
+    s.dpath = m = ProcDpath()
 
-      # dmem ports
-      dmemreq_addr  = s.dmem.req.msg.addr,
-      dmemreq_data  = s.dmem.req.msg.data,
-      dmemresp_data = s.dmemresp_q.deq.ret.data,
+    # imem ports
+    m.imemreq_addr  //= s.imemreq_q.enq.msg.addr
```
Comment (ptpan) [inline examples/ex03_proc/ProcRTL.py]: I imagine there could be a `stdlib.inspect_connect`  function that inspects two components and connect their ports based on the names... would be even better than SV `.*`

File: pymtl3/stdlib/rtl/enrdy_queues_test.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 import pytest
 
 from pymtl3 import *
-from pymtl3.stdlib.test import TestSinkCL, TestSrcCL
+from pymtl3.stdlib.test import TestSinkCL, TestSrcCL, run_sim
```
Comment (ptpan) [inline pymtl3/stdlib/rtl/enrdy_queues_test.py]: Is the idea to prefer using `stdlib.run_sim` over customized ticking method? That sounds like a huge improvement

Comment (jsn1993) [reply pymtl3/stdlib/rtl/enrdy_queues_test.py]: I think the problem is we had too many run_sim with basically the same code ... Now I just remove the redundant ones. I already did that in pymtl3-net.

File: examples/ex03_proc/ProcRTL.py (hunk 1 above)
Comment (jsn1993) [reply examples/ex03_proc/ProcRTL.py]: connect_auto in pymtl2 basically ..

**Pull request merged**